SQF-VM/SQF_types.c

```diff
@@ -327,6 +327,66 @@ void array_push(PARRAY arr, VALUE val)
 	arr->data[arr->top]->val = val.val;
 	arr->top++;
 };
+
+PVALUE array_pop(PARRAY array) {
+	array->top--;
+	PVALUE val = array->data[array->top];
+	array->data[array->top] = NULL;
+
+	return val;
+}
+
+PVALUE array_popAt(PARRAY array, int index) {
+	if(index < 0 || index >= array->top) {
+		return value_create(NOTHING_TYPE(), base_int(0));
+	}
+
+	PVALUE val = array->data[index];
+
+	for(int i=index; i<array->top - 1; i++) {
+		array->data[i] = array->data[i+1];
+	}
+
+	array->top--;
+	array->data[array->top] = NULL;
+
+	return val;
+}
+
+void array_resizeSQF(PARRAY array, int newSize) {
+	if(array->top > newSize) {
+		// delete array elements until respective size is reached
+		for(int i=array->top - 1; i>=newSize; i--) {
+			inst_destroy_value(array->data[i]);
+			array->data[i] = NULL;
+		}
+	} else {
+		// create respective elements with nil value
+		for(int i=array->top; i < newSize; i++) {
+			array_push(array, value(NOTHING_TYPE(), base_int(0)));
+		}
+	}
+
+	array->top = newSize;
+}
+
+/*
+ * Appends array2 to array1
+ */
+void array_append(PARRAY array1, PARRAY array2) {
+	for(int i=0; i<array2->top; i++) {
+		array_push(array1, *(value_copy(array2->data[i])));
+	}
+}
+
+void array_reverse(PARRAY array) {
+	PARRAY copy = array_copy(array);
```

Comment:
memory leak as copy is never cleared,
speed can be improved to `O(x/2)` by using two indexes meeting in the middle

alternative:
```c
void array_reverse(PARRAY array)
{
	int i, j;
	PVALUE tmp;
	for (i = 0, j = array->top - 1; i < j; i++, j--)
	{
		tmp = array->data[i];
		array->data[i] = array->data[j];
		array->data[j] = tmp;
	}
}
```
please note upper code is untested